### ISSUE TITLE
Compute custom video with COMEDIAN

### ISSUE THREAD

**Issue by edugzlez** [juliendenize/eztorch#5]
I am interested in computing my own video with COMEDIAN to detect soccer events.

Specifically I would like to use the already pretrained model for a number of different events.

I have tried to capture the model in a Python script to load the model.

```python
from hydra import compose, initialize
import hydra
from eztorch.utils.utils import compile_model

# initialize

with initialize(
    config_path="./eztorch/configs/run/finetuning/vivit",
    version_base="1.11",
):
    config = compose(config_name="vivit_tiny_soccernet_uniform").
    model = hydra.utils.instantiate(config.model)
    model = compile_model(model, config)
```

Load .pth:
```python
ckpt_path = "./comedian_vivit_tiny_seed203.pth"
state_dict = torch.load(ckpt_path)
state_dict["train_transform.0._transform.0.transforms.5.mean"] = state_dict["train_transform.0._transform.0.transforms.5.mean"].view(3, 1, 1)
state_dict["train_transform.0._transform.0.transforms.5.std"] = state_dict["train_transform.0._transform.0.transforms.5.std"].view(3, 1, 1)
model._orig_mod.load_state_dict(state_dict, strict=False)
```

The result:
```
_IncompatibleKeys(missing_keys=['train_transform.0._transform.0.transforms.2.brightness', 'train_transform.0._transform.0.transforms.2.contrast', 'train_transform.0._transform.0.transforms.2.saturation', 'val_transform.1._transform.0.transforms.2.mean', 'val_transform.1._transform.0.transforms.2.std', 'test_transform.1._transform.0.transforms.2.mean', 'test_transform.1._transform.0.transforms.2.std'], unexpected_keys=['trunk.transformer.temporal_mask_token', 'trunk.transformer.temporal_transformer.blocks.4.norm1.weight', 'trunk.transformer.temporal_transformer.blocks.4.norm1.bias', 'trunk.transformer.temporal_transformer.blocks.4.attn.qkv.weight', 'trunk.transformer.temporal_transformer.blocks.4.attn.qkv.bias', 'trunk.transformer.temporal_transformer.blocks.4.attn.proj.weight', 'trunk.transformer.temporal_transformer.blocks.4.attn.proj.bias', 'trunk.transformer.temporal_transformer.blocks.4.norm2.weight', 'trunk.transformer.temporal_transformer.blocks.4.norm2.bias', 'trunk.transformer.temporal_transformer.blocks.4.mlp.fc1.weight', 'trunk.transformer.temporal_transformer.blocks.4.mlp.fc1.bias', 'trunk.transformer.temporal_transformer.blocks.4.mlp.fc2.weight', 'trunk.transformer.temporal_transformer.blocks.4.mlp.fc2.bias', 'trunk.transformer.temporal_transformer.blocks.5.norm1.weight', 'trunk.transformer.temporal_transformer.blocks.5.norm1.bias', 'trunk.transformer.temporal_transformer.blocks.5.attn.qkv.weight', 'trunk.transformer.temporal_transformer.blocks.5.attn.qkv.bias', 'trunk.transformer.temporal_transformer.blocks.5.attn.proj.weight', 'trunk.transformer.temporal_transformer.blocks.5.attn.proj.bias', 'trunk.transformer.temporal_transformer.blocks.5.norm2.weight', 'trunk.transformer.temporal_transformer.blocks.5.norm2.bias', 'trunk.transformer.temporal_transformer.blocks.5.mlp.fc1.weight', 'trunk.transformer.temporal_transformer.blocks.5.mlp.fc1.bias', 'trunk.transformer.temporal_transformer.blocks.5.mlp.fc2.weight', 'trunk.transformer.temporal_transformer.blocks.5.mlp.fc2.bias', 'val_transform.1._transform.2.mean', 'val_transform.1._transform.2.std', 'test_transform.1._transform.2.mean', 'test_transform.1._transform.2.std'])
```

I think the pth has loaded well minus a few transforms. I'm not sure if I'm performing the model loading in the best way: I'm trying to use it freely from Python, because when I train it I want to use it in a process.

If I am right the `model._orig_mod.trunk` layer directly extracts the features every 2 frames.

```python
x = torch.randn(1, 3, 128, 224, 224).cuda()
model.eval()
model = model.cuda()
with torch.no_grad():
    feats = model._orig_mod.trunk(x)
    y = model(x)
    assert torch.allclose(feats, y['h'], atol=1e-5)
```

Please confirm me these questions and if I am doing it right.

**Comment by juliendenize** [comment]
Hello @edugzlez, thanks for taking an interest in our work.

I think the model loaded properly, it might have been a better idea on my part to remove from the `state_dict` the transformation parameters as they are not pertinent and should be config-based. I think that you loaded it right.

Still, I would suggest that you filter the transform keys from the model state_dict if you do not wish to use any transforms. You can do something like this (I did not proofread it but I hope you get the idea):
```py
config.model.train_transform=None # Remove train_transform initialization
config.model.val_transform=None # Remove val_transform initialization
config.model.test_transform=None # Remove test_transform initialization

model = ....

state_dict = torch.load(...)
filtered_state_dict = state_dict.copy()
for key in state_dict.keys():
   if key.startswith(("train_transform","val_transform","test_transform")): # Remove transforms from state_dict
      del filtered_state_dict[key]
model.load_state_dict(filtered_state_dict) # Only load model parameters
```

However, keep in mind you would have to resize and normalize the input data before passing them to the model.

> If I am right the model._orig_mod.trunk layer directly extracts the features every 2 frames.

This is right indeed you should obtain 64 tokens at the output if you pass 128 frames.

Did I answer correctly to your questions?

**Comment by edugzlez** [comment]
Yes @juliendenize, thank you very much. I have already been able to use the pre-trained network.

Thank you very much and congratulations on your results.

**Comment by tringuyen98** [comment]
Hi @edugzlez, can you provide me steps to inference a soccer video file to detect soccer events, many thanks.

**Comment by arshDeepTech** [comment]
@juliendenize @edugzlez  Can you guys help with the inference code and instructions for running on a custom video? The mentioned instructions seemed to not work.
